nocasedict/_nocasedict.py

```diff
@@ -0,0 +1,475 @@
+    # Other stuff
+
+    def __repr__(self):
+        """
+        Return a string representation of the object that is suitable for
+        debugging.
+
+        The order of dictionary items in the result is the preserved order of
+        adding or deleting items.
+
+        The lexical case of the keys in the result is the preserved lexical
+        case.
+        """
+        items = ["{0!r}: {1!r}".format(key, value)
+                 for key, value in self.iteritems()]
+        items_str = ', '.join(items)
+        return "{0.__class__.__name__}({{{1}}})".format(self, items_str)
+
+    def update(self, *args, **kwargs):
+        """
+        Update the dictionary from key/value pairs. If an item for a key
+        exists in the dictionary, its value is updated. If an item for a key
+        does not exist, it is added to the dictionary at the end.
+        The provided keys and values are stored in the dictionary without
+        being copied.
+
+        Each positional argument can be:
+
+          * an object with a method `items()` that returns an
+            :term:`py:iterable` of tuples containing key and value.
+
+          * an object without such a method, that is an :term:`py:iterable` of
+            tuples containing key and value.
+
+        Each keyword argument is a key/value pair.
+
+        The updates are performed first for the positional arguments in the
+        iteration order of their iterables, and then for the keyword arguments.
+        Note that before Python 3.4, keyword arguments are passed to this
+        method as a standard dict, so the order of updates for the keyword
+        arguments is not preserved.
+        """
+        for mapping in args:
+            if hasattr(mapping, 'items'):
+                for key, value in mapping.items():
+                    self[key] = value
+            else:
+                for key, value in mapping:
+                    self[key] = value
+        for key, value in kwargs.items():
+            self[key] = value
+
+    def clear(self):
+        """
+        Remove all items from the dictionary.
+        """
+        self._data.clear()
+
+    def copy(self):
+        """
+        Return a copy of the dictionary.
+
+        This is a middle-deep copy; the copy is independent of the original in
+        all attributes that have mutable types except for:
+
+        * The values in the dictionary
+
+        Note that the Python functions :func:`py:copy.copy` and
+        :func:`py:copy.deepcopy` can be used to create completely shallow or
+        completely deep copies of objects of this class.
+        """
+        result = NocaseDict()
+        result._data = self._data.copy()  # pylint: disable=protected-access
+        return result
+
+    def __eq__(self, other):
+        """
+        Invoked when two dictionaries are compared with the `==` operator.
+
+        The comparison is based on matching key/value pairs.
+        The keys are looked up case-insensitively.
```

Comment:
Again "Key lookup is case-insensitive.

Reply:
Has already been improved in docs update PR, please review there again.